libdnf5/base/goal.cpp

```diff
@@ -604,7 +604,9 @@ GoalProblem Goal::Impl::resolve_group_specs(std::vector<GroupSpec> & specs, base
     comps::GroupQuery base_groups_query(base);
     comps::EnvironmentQuery base_environments_query(base);
     for (auto & [action, reason, spec, settings] : specs) {
-        bool skip_unavailable = settings.resolve_skip_unavailable(cfg_main);
+        // For REMOVE action set skip_unavailable always to true
+        // to not treat the removal of a not-installed group as an error
+        bool skip_unavailable = action == GoalAction::REMOVE ? true : settings.resolve_skip_unavailable(cfg_main);
```

Comment:
My apology, but I am thinking whether it is a good approach to hard code the behavior. `skip_unavailable` has 3 options - `TRUE`, `FALSE`, `AUTO`.

What about to add additional condition to `action == GoalAction::REMOVE` and check whether  `settings.skip_unavailable == GoalSetting::AUTO`? And only if both conditions are 
`true` then use `skip_unavailable=true`, otherwise use `settings.resolve_skip_unavailable(cfg_main)`

This change will allow API users to set behavior according to their expectations, but will keep reasonable default.

Reply:
OK, that make sense. I'll change also package removal accordingly.

Reply:
Done.